root-parent-pom/pom.xml

```diff
@@ -406,7 +406,6 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <argLine>
-                                --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-exports=java.base/sun.nio.ch=ALL-UNNAMED
```

Comment:
why remove?

Reply:
I am also curious why..,

Reply:
because it exists in 11?

Reply:
That sounds fair. If it works then I approve!

Reply:
I don't think it works that way....

`mvn help:effective-pom -Pjava11,java17` results in
```
<plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>3.0.0-M4</version>
        <configuration>
          <argLine>--add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
                                --add-opens=jdk.compiler/com.sun.tools.javac=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
                                --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
                                --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED</argLine>
        </configuration>
      </plugin>
```
while
`mvn help:effective-pom -Pjava11` results in
```
<plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>3.0.0-M4</version>
        <configuration>
          <argLine>--add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-exports
                                java.base/jdk.internal.misc=ALL-UNNAMED --add-exports java.base/sun.nio.ch=ALL-UNNAMED</argLine>
        </configuration>
      </plugin>
```
and `mvn help:effective-pom -Pjava17` results in
```
<plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>3.0.0-M4</version>
        <configuration>
          <argLine>--add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
                                --add-opens=jdk.compiler/com.sun.tools.javac=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
                                --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
                                --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED</argLine>
        </configuration>
      </plugin>
```

You may be able to make it work that way with `combine.children=append`, but perhaps it's better just to duplicate it incase the profiles are explicitly specified and the caller doesn't realise they need to enable both?

Reply:
See https://blog.sonatype.com/2011/01/maven-how-to-merging-plugin-configuration-in-complex-projects/

Reply:
That's 11 years old though... so I wonder if it's still correct, Maven moves pretty slowly

Reply:
The caller should not need to enable any more

Reply:
But you can explicitly set which profiles are enabled I think... which would override the JDK detection.